cli/azd/internal/repository/app_init.go

```diff
@@ -560,54 +560,81 @@ func (i *Initializer) prjConfigFromDetect(
 			})
 
 		for _, database := range databases {
+			var resourceConfig project.ResourceConfig
+			var databaseName string
 			if database == appdetect.DbRedis {
-				redis := project.ResourceConfig{
-					Type: project.ResourceTypeDbRedis,
-					Name: "redis",
+				databaseName = "redis"
+			} else {
+				var err error
+				databaseName, err = i.getDatabaseNameByPrompt(ctx, database)
+				if err != nil {
+					return config, err
+				}
+			}
+			var authType = internal.AuthTypeUnspecified
+			if database == appdetect.DbPostgres || database == appdetect.DbMySql {
+				var err error
+				authType, err = chooseAuthTypeByPrompt(
+					databaseName,
+					[]internal.AuthType{internal.AuthTypeManagedIdentity, internal.AuthTypePassword},
+					ctx,
+					i.console)
+				if err != nil {
+					return config, err
 				}
-				config.Resources[redis.Name] = &redis
-				dbNames[database] = redis.Name
-				continue
 			}
-
-			var dbType project.ResourceType
 			switch database {
+			case appdetect.DbRedis:
+				resourceConfig = project.ResourceConfig{
+					Type: project.ResourceTypeDbRedis,
+					Name: databaseName,
+				}
 			case appdetect.DbMongo:
-				dbType = project.ResourceTypeDbMongo
-			case appdetect.DbPostgres:
-				dbType = project.ResourceTypeDbPostgres
-			case appdetect.DbMySql:
-				dbType = project.ResourceTypeDbMySQL
+				resourceConfig = project.ResourceConfig{
+					Type: project.ResourceTypeDbMongo,
+					Name: databaseName,
+					Props: project.MongoDBProps{
+						DatabaseName: databaseName,
+					},
+				}
 			case appdetect.DbCosmos:
-				dbType = project.ResourceTypeDbCosmos
-			}
-
-			db := project.ResourceConfig{
-				Type: dbType,
-			}
-
-			for {
-				dbName, err := promptDbName(i.console, ctx, database)
-				if err != nil {
-					return config, err
+				resourceConfig = project.ResourceConfig{
+					Type: project.ResourceTypeDbCosmos,
+					Name: databaseName,
+					Props: project.CosmosDBProps{
+						DatabaseName: databaseName,
```

Comment:
Is this database name the name of the database server, or a database?

Reply:
It's name of database. Now we do not have the information of database server name. The server name is '${abbrs.dBforMySQLServers}${resourceToken}'.  How about using const string for each database? For example: for cosmos db, use "cosmosdb", for MySql, use "mysql"

Reply:
Updated.

Reply:
No need to update this, let's keep the random name now for some services have constraints on global unique service name. I was just curious about where the database name is used for.